File: apps/fluux/src-tauri/check-tauri-features.py
#!/usr/bin/env python3
"""Guard the focus fix decision: the app must not opt into Tauri's unstable webview mode."""

from pathlib import Path
import sys
import tomllib


manifest = Path(__file__).with_name("Cargo.toml")
with manifest.open("rb") as stream:
    features = tomllib.load(stream)["dependencies"]["tauri"]["features"]

if "unstable" in features:
    print("tauri/unstable must remain disabled for the single-webview focus path", file=sys.stderr)
    raise SystemExit(1)

print("tauri/unstable is disabled")

File: apps/fluux/src-tauri/check-tauri-features.mjs
import { readFileSync } from 'node:fs'
import TOML from '@iarna/toml'

const manifest = TOML.parse(readFileSync(new URL('./Cargo.toml', import.meta.url), 'utf8'))
const dependencies = [manifest.dependencies, ...Object.values(manifest.target ?? {}).map(target => target.dependencies)]
const features = dependencies.flatMap(deps => deps?.tauri?.features ?? [])

if (features.includes('unstable')) {
  console.error('tauri/unstable must remain disabled for the single-webview focus path')
  process.exit(1)
}

console.log('tauri/unstable is disabled')
